ecs/client.go

```diff
@@ -300,3 +306,43 @@ func (c *BasicECSClient) isNonRetryableErrorCode(code string) bool {
 		return false
 	}
 }
+
+// isTaskNotFoundError returns whether or not the error returned from ECS is
+// because the task cannot be found.
+func isTaskNotFoundError(err error) bool {
+	awsErr, ok := err.(awserr.Error)
+	if !ok {
+		return false
+	}
+	return awsErr.Code() == ecs.ErrCodeInvalidParameterException && strings.Contains(awsErr.Message(), "The referenced task was not found")
+}
+
+// ConvertFailureToError converts an ECS failure message into a formatted error.
+// If the failure is due to being unable to find the task, it will return a
+// cocoa.ECSTaskNotFound error.
+// Docs: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/api_failures_messages.html
+func ConvertFailureToError(f ecs.Failure) error {
+	if isTaskNotFoundFailure(f) {
+		return cocoa.NewECSTaskNotFoundError(utility.FromStringPtr(f.Arn))
+	}
+	var parts []string
+	if arn := utility.FromStringPtr(f.Arn); arn != "" {
+		parts = append(parts, fmt.Sprintf("task '%s'", arn))
+	}
+	if reason := utility.FromStringPtr(f.Reason); reason != "" {
+		parts = append(parts, fmt.Sprintf("(reason) %s", reason))
+	}
+	if detail := utility.FromStringPtr(f.Detail); detail != "" {
+		parts = append(parts, fmt.Sprintf("(detail) %s", detail))
+	}
+	if len(parts) == 0 {
+		return errors.New("ECS failure did not contain any additional failure information")
```

Comment:
Is it necessary to fail here? Seems like we could no-op

Reply:
I don't really have feelings either way about no-op vs. erroring here, but I can change it to no-op if you think it's better. This is a "not supposed to happen" kind of case, because we assume that if they give us a non-nil failure, then it must have at least some information populated. I only added this to avoid the possibility that L341 can return an unhelpful error that's missing a message.

Reply:
Hmm I guess it's better to fail if the `Failures` field has anything at all so I'm ok keeping it this way.